test/cql-pytest/util.py

```diff
@@ -63,15 +63,38 @@ def format_tuples(tuples=None, **kwargs):
     body = ', '.join(f"'{key}': '{value}'" for key, value in tuples.items())
     return f'{{ {body} }}'
 
+def keyspace_has_tablets(cql, keyspace):
+    """ Return true if the test_keyspace was created with tablets.
+
+    We support running cql-pytest against an older version of scylla, so we do
+    the detection in a way that accounts for scylla possibly not even knowing
+    what tablets is.
+
+    If the keyspace was created with tablets, it will have an entry in
+    `system_schema.scylla_keyspaces`, with `initial_tablets` set to 0.
+    So here, we simply query this table, looking for a partition for the
+    appropriate keyspace. If the result has the `initial_tablets` column and it
+    is set, the keyspace has tablets.
+    """
+    # Need to use network strategy, otherwise tablets will not be enabled.
+    res = list(cql.execute(f"SELECT * FROM system_schema.scylla_keyspaces WHERE keyspace_name='{keyspace}'"))
+    if not res:
+        return False
+    return getattr(res[0], "initial_tablets", None) is not None
 
 # A utility function for creating a new temporary keyspace with given options.
 # It can be used in a "with", as:
 #   with new_test_keyspace(cql, '...') as keyspace:
 # This is not a fixture - see those in conftest.py.
+# The force_no_tablets options allows disabling tablets, if scylla would enable
+# it by default.
 @contextmanager
-def new_test_keyspace(cql, opts):
+def new_test_keyspace(cql, opts, force_no_tablets=False):
     keyspace = unique_name()
     cql.execute("CREATE KEYSPACE " + keyspace + " " + opts)
+    if force_no_tablets and keyspace_has_tablets(cql, keyspace):
+        cql.execute("DROP KEYSPACE " + keyspace)
+        cql.execute("CREATE KEYSPACE " + keyspace + " " + opts + " AND TABLETS = {'enabled': false}")
```

Comment:
Since in v2 there were only 3 callers left, for 2 `create_keyspace_*()` variants, I just extracted the logic into the callers instead. We shouldn't make it easy to disable tablets in a test.